spatialos-sdk/src/worker/entity.rs

```diff
@@ -36,6 +36,35 @@ impl Entity {
         Ok(entity)
     }
 
+    pub(crate) fn from_schema_field(object: &SchemaObject) -> Result<Self, schema::Error> {
+        let field_ids = object.unique_field_ids();
+        let mut entity = Entity::new();
+
+        for field_id in field_ids {
+            entity
+                .pre_add_check(field_id)
+                .map_err(Error::schema_error::<Self>)?;
+
+            let component = object.get_object(field_id);
+            let mut owned = SchemaComponentData::new();
+            owned
+                .fields_mut()
+                .copy_from(component)
+                .map_err(Error::schema_error::<Self>)?;
+
+            entity.components.insert(field_id, owned);
+        }
+
+        Ok(entity)
+    }
+
+    pub(crate) fn to_schema_field(&self, object: &mut SchemaObject) {
+        for (component_id, data) in &self.components {
+            let component_field = object.add_object(*component_id);
+            component_field.copy_from(data.fields()).unwrap(); // TODO: Get rid of unwrap.
```

Comment:
@dgavedissian @randomPoison 

How do we want to handle the result from `SchemaObject::copy_from`? 

The underlying schema operations are fallible, although the only possible error (currently) is that the Schema Object is greater than 2 GB in size.

Reply:
Huh, I'm surprised that the logic for copying a schema object is fallible when the logic for serializing into a schema object isn't 🤔 Is there anything wrong with returning a `Result` from `to_schema_field`?

Reply:
We can return a result from `to_schema_field` but it will simply be unwrapped/ignored at some point as serialization isn't fallible. 

> I'm surprised that the logic for copying a schema object is fallible when the logic for serializing into a schema object isn't 

Yeah this was confusing to me as well

Reply:
If we need to we can totally make serialization fallible. On the other hand, there's probably no way to meaningfully recover from being unable to allocate enough space for your schema objects. It's pretty similar to an out-of-memory error, which Rust generally handles by panicking. So maybe it'd be fine to panic if the schema data can't be copied?

Reply:
Yeah that's sort of my opinion too. I'm kind of okay with leaving in this `unwrap()`

Reply:
Maybe switch it to `expect` so that we can at least give some context on what's breaking when someone inevitably hits it 😛 Otherwise I'm good with having it panic 👍

Reply:
Also then satisfies the `TODO` comment :rocket:

Reply:
Hah, true! 🎉